src/trading/tradingSdk.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  LimitOrderAdvancedSettings,
+  LimitTradeParameters,
+  QuoteAndPost,
+  SwapAdvancedSettings,
+  TradeParameters,
+  TraderParameters,
+} from './types'
+import { postSwapOrder, postSwapOrderFromQuote } from './postSwapOrder'
+import { postLimitOrder } from './postLimitOrder'
+import { getQuoteWithSigner } from './getQuote'
+import { postSellNativeCurrencyTrade } from './postSellNativeCurrencyTrade'
+import { getSigner, swapParamsToLimitOrderParams } from './utils'
+import { getPreSignTransaction } from './getPreSignTransaction'
+
+export class TradingSdk {
+  constructor(public readonly traderParams: TraderParameters) {}
+
+  async getQuote(params: TradeParameters, advancedSettings?: SwapAdvancedSettings): Promise<QuoteAndPost> {
+    const quoteResults = await getQuoteWithSigner(this.mergeParams(params), advancedSettings)
+
+    return {
+      quoteResults: quoteResults.result,
+      postSwapOrderFromQuote: () => postSwapOrderFromQuote(quoteResults),
+    }
+  }
+
+  async postSwapOrder(params: TradeParameters, advancedSettings?: SwapAdvancedSettings): Promise<string> {
+    return postSwapOrder(this.mergeParams(params), advancedSettings)
+  }
+
+  async postLimitOrder(params: LimitTradeParameters, advancedSettings?: LimitOrderAdvancedSettings): Promise<string> {
+    return postLimitOrder(this.mergeParams(params), advancedSettings)
+  }
+
+  async postSellNativeCurrencyTrade(
```

Comment:
Nitpick/question: why is this suffixed `Trade`, while the other 2 methods are suffixed `Order`?

Reply:
No big reasons, renamed it